core/textile/client.go

```diff
@@ -197,6 +199,12 @@ func (tc *textileClient) checkHubConnection(ctx context.Context) error {
 			return err
 		}
 		tc.mb = mailbox
+
+		if err := tc.listenForMessages(hubctx); err != nil {
+			tc.isConnectedToHub = false
```

Comment:
@dmerrill6 

Do you think we need any locking here?

Reply:
Will merge to unblock the FE and we can iterate if we find issues

Reply:
@dmerrill6 

Yes, let's discuss this later.